api/clients/v2/relay_payload_retriever.go

```diff
@@ -187,12 +174,24 @@ func (pr *RelayPayloadRetriever) GetPayload(
 func (pr *RelayPayloadRetriever) getBlobWithTimeout(
 	ctx context.Context,
 	relayKey core.RelayKey,
-	blobKey *core.BlobKey) ([]byte, error) {
+	blobKey *core.BlobKey,
+	blobLengthSymbols uint32,
+) (*codecs.Blob, error) {
 
 	timeoutCtx, cancel := context.WithTimeout(ctx, pr.config.RelayTimeout)
 	defer cancel()
 
-	return pr.relayClient.GetBlob(timeoutCtx, relayKey, *blobKey)
+	blobBytes, err := pr.relayClient.GetBlob(timeoutCtx, relayKey, *blobKey)
+	if err != nil {
+		return nil, fmt.Errorf("get blob from relay: %w", err)
+	}
+
+	blob, err := codecs.DeserializeBlob(blobBytes, blobLengthSymbols)
```

Comment:
should relayClient not returned a deserializedBlob? Or is this also one of the TODOs that you have somewhere in here?

Reply:
It should, add TODOs to indicate places I previously missed, where future work is needed [76dac208](https://github.com/Layr-Labs/eigenda/pull/1305/commits/76dac208bb98f9572610a72f75a0dad20b95335d)